app/assets/javascripts/discourse/app/mixins/uppy-upload.js

```diff
@@ -73,7 +73,6 @@ export default Mixin.create(UppyS3Multipart, ExtendableUploader, {
     });
     this.set("allowMultipleFiles", this.fileInputEl.multiple);
     this.set("inProgressUploads", []);
-    this._triggerInProgressUploadsEvent();
```

Comment:
@martin-brennan do you remember why we need to do this on insert? There's technically be 0 upload event at this moment

Reply:
Yes we should keep this here, it should be called every time `inProgressUploads` is changed even when it is set back to `[]`:

https://github.com/discourse/discourse/blob/e4d628a931cd9af1a37ce4f269bc691b61e10632/app/assets/javascripts/discourse/app/mixins/uppy-upload.js#L320-L326

IIRC I had to do this originally because of how uploads were passed down through drafts and chat composer. We actually do still use this, I checked:

https://github.com/discourse/discourse/blob/e4d628a931cd9af1a37ce4f269bc691b61e10632/plugins/chat/assets/javascripts/discourse/components/chat-composer.hbs#L110-L110

https://github.com/discourse/discourse/blob/e4d628a931cd9af1a37ce4f269bc691b61e10632/plugins/chat/assets/javascripts/discourse/components/chat-composer-uploads.js#L127-L137

So yes we should keep it, but agree it's kind of weird and could be refactored.